### PR TITLE
Add crew manifest action for borgs

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

Well, I think the Borg should have access to the crew's manifest, so yes, this will add a new action for them that will allow them to look at the manifest.

## Why / Balance

Idk.

## Technical details

Idk.

## Media

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: VigersRay
- tweak: Borgs can view station crew manifest.

### CODE REVIEW

**Comment by VigersRay** [comment]
yes, the icon is shit, but I'm not an artist.

**Comment by lzk228** [comment]
fixes #21835

**Review by lzk228 — CHANGES_REQUESTED** [review]
dont make locale strings for entity name and desc, translate could be done by downstreams with
`ent-action = name
    .desc = desc`

**Comment by VigersRay** [comment]
> fixes #21835

Well, in general, the idea with a PDA is not so bad. But then it’s better to make a separate UI in which the name of the station, time and code will be written and there will be a button to open the manifest as in the ID card console, but there will be a lot of copy-paste from the PDA.

**Comment by icekot8** [comment]
I wish pai had this feature :pray:

**Comment by VigersRay** [comment]
> I wish pai had this feature 🙏

By the way, I did it just for him, but for some reason I did it for the Borg. I would also like to see a health analyzer for it.

File: Content.Server/CrewManifest/CrewManifestSystem.cs [hunk 1]
```diff
@@ -46,6 +49,42 @@ public override void Initialize()
 
         SubscribeLocalEvent<CrewManifestViewerComponent, BoundUIClosedEvent>(OnBoundUiClose);
         SubscribeLocalEvent<CrewManifestViewerComponent, CrewManifestOpenUiMessage>(OpenEuiFromBui);
+        SubscribeLocalEvent<BorgCrewManifestViewerComponent, ComponentStartup>(BorgCrewManifestViewerStartup);
+        SubscribeLocalEvent<BorgCrewManifestViewerComponent, CrewManifestOpenActionEvent>(OpenCrewManifest);
+    }
+
+    private void BorgCrewManifestViewerStartup(EntityUid uid, BorgCrewManifestViewerComponent component, ComponentStartup args)
```
Comment (TheShuEd) [inline Content.Server/CrewManifest/CrewManifestSystem.cs:56]: Use new entity<>

File: Content.Server/CrewManifest/CrewManifestSystem.cs [hunk 2]
```diff
@@ -46,6 +49,42 @@ public override void Initialize()
...
+    {
+        _actions.AddAction(uid, component.ActionViewCrewManifest);
+    }
+
+    private void OpenCrewManifest(EntityUid uid, BorgCrewManifestViewerComponent component, CrewManifestOpenActionEvent args)
+    {
```
Comment (TheShuEd) [inline Content.Server/CrewManifest/CrewManifestSystem.cs:62]: Use new entity<>

File: Content.Shared/Silicons/Borgs/Components/BorgCrewManifestViewerComponent.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Shared.Silicons.Borgs.Components;
+
+[RegisterComponent]
+public sealed partial class BorgCrewManifestViewerComponent : Component
+{
+    [ViewVariables(VVAccess.ReadWrite),
+     DataField("actionViewCrewManifest", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string ActionViewCrewManifest = "ActionViewCrewManifest";
```
Comment (TheShuEd) [inline Content.Shared/Silicons/Borgs/Components/BorgCrewManifestViewerComponent.cs:11]: Use EntProtoId

File: Content.Shared/Silicons/Borgs/Components/BorgCrewManifestViewerComponent.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Shared.Silicons.Borgs.Components;
+
+[RegisterComponent]
+public sealed partial class BorgCrewManifestViewerComponent : Component
+{
+    [ViewVariables(VVAccess.ReadWrite),
+     DataField("actionViewCrewManifest", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
```
Comment (TheShuEd) [inline Content.Shared/Silicons/Borgs/Components/BorgCrewManifestViewerComponent.cs:10]: Remove customTypeSerializer
Remove naming

File: Resources/Textures/Interface/Actions/actions_borg.rsi/meta.json [hunk 5]
```diff
@@ -9,6 +9,9 @@
     "states": [
         {
             "name": "state-laws"
+        },
+        {
+            "name": "state-manifest"
```
Comment (TheShuEd) [inline Resources/Textures/Interface/Actions/actions_borg.rsi/meta.json:14]: Add license to new image

**Review by TheShuEd — CHANGES_REQUESTED** [review]
🤔

**Pull request closed without merging**